project/Build.scala

```diff
@@ -1440,6 +1459,73 @@ object Build {
       settings(commonBenchmarkSettings).
       enablePlugins(JmhPlugin)
 
+    def asScala3doc: Project = {
+      def generateDocumentation(targets: String, name: String, outDir: String, params: String = "") = Def.taskDyn {
+          val sourceMapping = "=https://github.com/lampepfl/dotty/tree/master#L"
+          run.in(Compile).toTask(s""" -o output/$outDir -t $targets -n "$name" -s $sourceMapping $params""")
```

Comment:
scalac/scaladoc/dotc/dottydoc have always used `-d` for the output directory, not `-o`: https://github.com/lampepfl/dotty/blob/d09600de025e6760dc8614a4ce62d568e90d16d5/compiler/src/dotty/tools/dotc/config/ScalaSettings.scala#L24

Reply:
I've tried to do that here: https://github.com/romanowski/dotty/commit/b9ac86d813b053915bd5b06dff283c7b1574cdd6 however there are few problems:
- inside compiler all arguments seems to be optional, so I am forced to wite pice of code to enforce that
- aliases exists in API however it seems that are not implemented

Since this is non-trivial task (especially that we will need to add test and refator a bit our API) maybe we can do it after initial merge (#10091 broke this PR and I will need to recreate this PR again)? I've created issue to track that: https://github.com/lampepfl/scala3doc/issues/223

Reply:
> Since this is non-trivial task (especially that we will need to add test and refator a bit our API) maybe we can do it after initial merge

Yes, that's fine.